x/wasm/types/proposal.go

```diff
@@ -852,6 +852,17 @@ func (p PinCodesProposal) ValidateBasic() error {
 	if len(p.CodeIDs) == 0 {
 		return sdkerrors.Wrap(ErrEmpty, "code ids")
 	}
+	for _, num1 := range p.CodeIDs {
+		if num1 == 0 {
+			return sdkerrors.Wrap(ErrZero, "0 is not accepted")
+
+		}
+		for _, num2 := range p.CodeIDs {
+			if num1 == num2 {
+				return sdkerrors.Wrap(ErrDuplicate, "duplicate")
```

Comment:
The value would be a good error message for the duplicate type. What do you think?

Reply:
yeah got it i will add it now